core/src/main/java/org/apache/accumulo/core/util/LocalityGroupUtil.java

```diff
@@ -340,4 +340,14 @@ public static void seek(FileSKVIterator reader, Range range, String lgName, Map<
 
     reader.seek(range, families, inclusive);
   }
+
+  static public void ensureNonOverlappingGroups(Map<String,Set<Text>> groups) {
+    HashSet<Text> all = new HashSet<>();
+    for (Entry<String,Set<Text>> entry : groups.entrySet()) {
+      if (!Collections.disjoint(all, entry.getValue())) {
+        throw new IllegalArgumentException("Group " + entry.getKey() + " overlaps with another group");
+      }
+      all.addAll(entry.getValue());
+    }
+  }
```

Comment:
This location would make this public API... but I'm not sure we want to do that for an internal utility method.

Reply:
I had not thought of the public api implications when refactoring. I was trying to consolidate some code. I can revert the functionality back into TableOperations and then add a separate check within NewTableConfiguration or create a static method in TableOperationsHelper as @keith-turner  suggested for the checkIteratorConflict method. Thoughts?

Reply:
Oh, my mistake. I thought this was `o.a.a.core.client.util`; it's actually `o.a.a.core.util`, which is not public API. Never mind :smile: